src/IO/AsynchronousReader.h

```diff
@@ -61,6 +61,8 @@ class IAsynchronousReader : private boost::noncopyable
         /// offset
         /// Optional. Useful when implementation needs to do ignore().
         size_t offset = 0;
+
+        operator std::tuple<size_t &, size_t &>() { return {size, offset}; }
```

Comment:
`std::pair` would be more appropriate

Reply:
It will not work with `std::pair` - compile error `error: no viable overloaded '='`